vortex-serde/src/layouts/read/layouts/chunked.rs

```diff
@@ -75,6 +95,69 @@ impl ChunkedLayout {
         }
     }
 
+    fn child_ranges(&self) -> VortexResult<Vec<(usize, usize)>> {
+        let MetadataState::Array(ref m) = self.metadata_array else {
+            vortex_bail!("Must fetch metadata before")
+        };
+
+        let row_offset = m
+            .with_dyn(|a| a.as_struct_array_unchecked().field(0))
+            .ok_or_else(|| vortex_err!("must have row_offset metadata column"))?;
+        let primitive_offsets = row_offset.into_primitive()?;
+        Ok(primitive_offsets
+            .maybe_null_slice::<u64>()
+            .iter()
+            .chain(&[self.length])
+            .tuple_windows()
+            .map(|(begin, end)| (*begin as usize, *end as usize))
+            .collect::<Vec<_>>())
+    }
+
+    fn ranged_children(&self) -> VortexResult<VecDeque<RangedLayoutReader>> {
+        let dtype = self.message_cache.dtype();
+        self.flatbuffer()
+            .children()
+            .ok_or_else(|| vortex_err!("Missing children"))?
+            .iter()
+            .enumerate()
```

Comment:
should this be after the `skip`  to ensure index always starts at 0?

Reply:
i'm not sure if the message_cache is inclusive of the metadata child

Reply:
The metadata child is a message like any other one so we want stable indexes and metadata would be 0